src/libos/src/process/do_exec.rs

```diff
@@ -21,30 +23,46 @@ pub fn do_exec(
     argv: &[CString],
     envp: &[CString],
     current_ref: &ThreadRef,
+    context: *mut CpuContext,
 ) -> Result<isize> {
     trace!(
         "exec current process pid = {:?}",
         current_ref.process().pid()
     );
 
-    // Construct new process structure but with same parent, pid, tid
-    let current = current!();
-    let new_process_ref = super::do_spawn::new_process_for_exec(path, argv, envp, current_ref);
+    let (is_forked, tid, parent_process) = check_vfork_for_exec(current_ref);
```

Comment:
Maybe (is_forked, tid, parent_process) could be replaced with Option(tid, parent_process)

Reply:
Done. Thanks for your review.